### DISCUSSION TITLE
Difference between revalidate and mutate

### DISCUSSION THREAD

**Discussion by andrehsu** [vercel/swr#812] [Q&A]
What is the difference between revalidate and mutate? The docs aren’t really clear, and revalidate isn't even mentioned at all.

**Comment by promer94** [comment]
***revalidate*** will be deprecated in the future.

**Reply by andrehsu** [reply]
`mutate` seems to
1. delete the old data
1. return nothing from `useSwr`
1. fetch, then
1. return the new data.

Whereas `revalidate` seems to 
1. return the old data from `useSwr`
1. fetch/delete old data, then
1. return the new data

The difference is that `revalidate` always returns some form of data for `useSwr` during the process, while `mutate` leaves a gap where no data is returned. Is there a way to replicate the behavior of `revalidate` using `mutate` when `revalidate` gets deprecated?

**Comment by promer94** [comment]
mutate will not delete the old data. calling mutate() is same
as calling revalidate()
